webapp/src/ts/modules/analytics/analytics-target-aggregates-sidebar-filter.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { Store } from '@ngrx/store';
+
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { UserSettingsService } from '@mm-services/user-settings.service';
+
+@Component({
+  selector: 'mm-analytics-target-aggregates-sidebar-filter',
+  templateUrl: './analytics-target-aggregates-sidebar-filter.component.html'
+})
+export class AnalyticsTargetAggregatesSidebarFilterComponent implements OnInit, OnDestroy {
```

Comment:
I found this error. 
Offline user gets an upgrade app popup and the app reloads while in targets aggregate page. So for example:
1. This user didn't have the targets aggregate enabled, so it displays `target aggregates are disabled`. 
2. Then admin enables, 
3. and the while the offline user is in the page with `target aggregates are disabled`, the user gets an upgrade popup, 
4. app reloads, and see error in browser console, and filer doesn't show. 
<img width="460" alt="Screenshot 2024-07-16 at 12 16 34 PM" src="https://github.com/user-attachments/assets/24c95616-e5d3-4927-94de-fc02dd02752b">
<img width="739" alt="Screenshot 2024-07-16 at 12 20 37 PM" src="https://github.com/user-attachments/assets/524ba3dc-0ef8-4469-b88e-c61580184f6b">

The filter should display with no errors, because the user already has the permission to access it

Reply:
Wait, this user has only 1 facility assigned. So it should not show the filter, but it should not throw an error. 
So it still needs a fix.

Reply:
Good catch. 


https://github.com/user-attachments/assets/f244060d-d775-4c0d-958e-c79a9ffece47

I have fixed it. Let me know if the video captures the fix

Reply:
Yeah, but also, if the user has only 1 user assigned, it shouldn't throw that error after reloading the app on upgrade.